Board.java

```diff
@@ -0,0 +1,634 @@
+    }, {
+            new BoardSquare(1, 3),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 3),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(2, 1),
+            new BoardSquare(),
+            new BoardSquare(),
+            new BoardSquare(1, 3),
+    }
+    };
+    private static Board SingleBoard = null;
+
+    private Board() {}
+
+    public static Board getBoard() {
+        if(SingleBoard == null) {
+            SingleBoard = new Board();
+        }
+        return SingleBoard;
+    }
+
+    public Tile[][] getTiles() {
+        Tile[][] tilesBoard = new Tile[BOARD_SIZE][BOARD_SIZE];
+
+        for (int i = 0; i < board[0].length; i++) {
+            for (int j = 0; j < board.length; j++) {
+                tilesBoard[i][j] = board[i][j].getTile();
+            }
+        }
+
+        return tilesBoard;
+    }
+
+    public boolean boardLegal(Word word) {
+        return isWordFitInBoard(word) && isWordLayOnOrNextToOther(word) && !isReplacingATile(word);
+    }
+
+    private boolean isWordFitInBoard(Word word) {
+
+        int row = word.getRow();
+        int col = word.getCol();
+
+        if(row < 0 || row > BOARD_SIZE - 1 || col < 0 || col > BOARD_SIZE - 1) {
+            return false;
+        }
+
+        int wordLength = word.getTiles().length;
+
+        if(word.isVertical()){
+            return row + wordLength <= BOARD_SIZE - 1;
+        }
+        return col + wordLength <= BOARD_SIZE - 1;
+    }
+
+    private boolean isReplacingATile(Word word) {
+        int row = word.getRow();
+        int col = word.getCol();
+        Tile[] wordTiles = word.getTiles();
+
+        if(word.isVertical()) {
+            for (int i = row; i < row + wordTiles.length; i++) {
+                if (wordTiles[i - row] != null && board[i][col].isOccupied()) {
+                        return true;
+                }
+            }
+        }
+        else {
+            for (int i = col; i < col + wordTiles.length; i++) {
+                if (wordTiles[i - col] != null && board[row][i].isOccupied()) {
+                        return true;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private boolean isWordLayOnOrNextToOther(Word word) {
+        int row = word.getRow();
+        int col = word.getCol();
+        int wordLength = word.getTiles().length;
+        int centerIndex = BOARD_SIZE / 2;
+
+        // Very much not efficient - but for now
+        boolean isFirstTurn = true;
+        for (int i = 0; i < board[0].length; i++) {
+            for (int j = 0; j < board.length; j++) {
+                if(board[i][j].isOccupied()) {
+                    isFirstTurn = false;
+                    break;
+                }
+            }
+        }
```

Comment:
i thought about adding another variable to the class, and when first updating the board, change it to true and then this check is O(1)

Reply:
sure, that's what I would do, but take it mind it's happening a single time so it doesn't really matter